### PR TITLE
Dave Gnukem

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/davidjoffe/dave_gnukem/pull/185
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)* https://github.com/davidjoffe/dave_gnukem/pull/185

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by Mailaender** [comment]
bot, build com.djoffe.gnukem

File: com.djoffe.gnukem.json [hunk 1]
```diff
@@ -0,0 +1,65 @@
+{
+  "app-id": "com.djoffe.gnukem",
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "22.08",
+  "sdk": "org.freedesktop.Sdk",
+  "command": "davegnukem",
+  "rename-desktop-file": "davegnukem.desktop",
+  "rename-icon-file": "davegnukem.png",
```
Comment (hfiguiere) [inline com.djoffe.gnukem.json:8]: this key doesn't exist.

File: metadata.patch [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+diff --git a/debian/appstream/com.djoffe.gnukem.metainfo.xml b/debian/appstream/com.djoffe.gnukem.metainfo.xml
+new file mode 100644
+index 0000000..62eae89
+--- /dev/null
++++ b/debian/appstream/com.djoffe.gnukem.metainfo.xml
+@@ -0,0 +1,67 @@
++<?xml version='1.0' encoding='utf-8'?>
++<component type="desktop">
++  <!--Created with jdAppdataEdit 5.1-->
++  <id>davegnukem</id>
```
Comment (hfiguiere) [inline metadata.patch:22]: please use the righ app-id. this is definitely not valid.

File: icon.patch [hunk 3]
```diff
@@ -0,0 +1,185 @@
+From 9ecfe5184377d2d0f7def946868c7d3e1ddc4794 Mon Sep 17 00:00:00 2001
```
Comment (hfiguiere) [inline icon.patch:1]: just add the icon file here. No need for this patch business.

File: com.djoffe.gnukem.json [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+      "sources":[
+        {
+          "type": "archive",
+          "url": "https://github.com/davidjoffe/dave_gnukem/releases/download/1.0.3/davegnukem_1.0.3.orig.tar.xz",
+          "sha256": "77a64e27ab8c006c8cab5cdec005a020dfe383e9aff35748ff9e426c2708d69c"
+        },
+        {
+          "type": "shell",
+          "commands": [
+            "sed -i 's#/usr/local#/app#g' Makefile"
```
Comment (hfiguiere) [inline com.djoffe.gnukem.json:44]: ideal you'd submit a patch to honour PREFIX. It's a matter of adding `?` in the right place. This kind of sed is error prone. And not honour prefix like that is a bug and should be fixed upstream. (assign with `?=`)

File: icon.patch (hunk 3 above)
Comment (hfiguiere) [reply icon.patch:1]: Unless your PR include updating the Makefile as I commented upstream.

File: com.djoffe.gnukem.json [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+      "name":"dave-gnukem",
+      "buildsystem":"autotools",
+      "no-autogen": true,
+      "no-make-install": true,
+      "post-install":
+      [
+        "install -Dvm755 davegnukem -t ${FLATPAK_DEST}/bin/",
+        "install -dv ${FLATPAK_DEST}/share/",
+        "cp -rfv debian/icons/ ${FLATPAK_DEST}/share/",
+        "install -Dm644 debian/desktop/davegnukem.desktop -t ${FLATPAK_DEST}/share/applications/",
```
Comment (hfiguiere) [inline com.djoffe.gnukem.json:30]: this is installed by upstream makefile

https://github.com/davidjoffe/dave_gnukem/blob/main/Makefile#L82

File: com.djoffe.gnukem.json (hunk 1 above)
Comment (hfiguiere) [reply com.djoffe.gnukem.json:8]:
```suggestion
  "rename-icon": "davegnukem",
```

**Comment by Mailaender** [comment]
bot, build com.djoffe.gnukem

File: com.djoffe.gnukem.json [hunk 6]
```diff
@@ -0,0 +1,62 @@
+{
+  "app-id": "com.djoffe.gnukem",
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "22.08",
+  "sdk": "org.freedesktop.Sdk",
+  "command": "davegnukem",
+  "rename-desktop-file": "davegnukem.desktop",
+  "rename-icon-file": "davegnukem",
```
Comment (hfiguiere) [inline com.djoffe.gnukem.json:8]:
```suggestion
  "rename-icon": "davegnukem",
```

I said the key didn't exist.

**Comment by Mailaender** [comment]
bot, build com.djoffe.gnukem

**Comment by Mailaender** [comment]
bot, build com.djoffe.gnukem

**Comment by barthalion** [comment]
LGTM, let's give upstream few more days to respond.

**Comment by davidjoffe** [comment]
Thanks so much!! If you don't mind, can we also invite @matteobin? That is if he wants to maybe contribute also, as he's also doing most the current work relating to this game's packaging for Debian etc.

**Pull request closed without merging**